### ISSUE TITLE
Property 'children' does not exist on type 'IntrinsicAttributes & DialogContentProps'.ts(2322)

### ISSUE THREAD

**Issue by aminya** [kobaltedev/kobalte#374]
**Describe the bug**
I am trying to use the `Dialog` components, but for many of them, TypeScript gives me an error: I cannot put children under the component. It seems the props of all these components inherit the form `AsChild`, which doesn't allow children.

The same issue exists for assigning class and ID to these components.

```ts
Type '{ id: string; }' is not assignable to type 'IntrinsicAttributes & DialogTriggerProps'.
  Property 'id' does not exist on type 'IntrinsicAttributes & DialogTriggerProps'
```

```ts
Type '{ children: Element; class: string; }' is not assignable to type 'IntrinsicAttributes & DialogCloseButtonProps'.
  Property 'children' does not exist on type 'IntrinsicAttributes & DialogCloseButtonProps'.
```

**To Reproduce**
Steps to reproduce the behaviour:

```tsx
  <Dialog.Root>
    <Dialog.Trigger id="error-trigger" />
    <Dialog.Portal>
      <Dialog.Overlay class="dialog__overlay" />
      <div class="dialog__positioner">
        <Dialog.Content class="dialog__content">
          <div class="dialog__header">
            <Dialog.Title class="dialog__title">Error</Dialog.Title>
            <Dialog.CloseButton class="dialog__close-button">
              <span innerHTML={cross} />
            </Dialog.CloseButton>
          </div>
          <Dialog.Description class="dialog__description">
            <div id="table-error" />
          </Dialog.Description>
        </Dialog.Content>
      </div>
    </Dialog.Portal>
  </Dialog.Root>
```

**Screenshots**

![image](https://github.com/kobaltedev/kobalte/assets/16418197/3cca6d2a-a805-4364-9022-cc4a8e5c4a48)

**Smartphone (please complete the following information):**

- Version     "@kobalte/core": "^0.12.6",

**Comment by aminya** [comment]
I have realized the issue is that the types in `@kobalte/core` depends on `@kobalte/utils`, but `@kobalte/utils` types are not being found by TypeScript in VsCode. Installing ` @kobalte/utils` as my dependency fixes the issue, but that is not ideal.

I am using `pnpm`.

**Comment by jer3m01** [comment]
We're working on fixing and changing out polymorphic and type system. 
If you'd like to follow progress or offer feedback, check out https://github.com/jer3m01/polymorphic or/and SolidJS' discord.

**Comment by jer3m01** [comment]
Types have been fixed in #381.
